web/src/activity_ui.ts

```diff
@@ -141,9 +143,22 @@ function do_update_users_for_search(): void {
     // Hide all the popovers but not userlist sidebar
     // when the user is searching.
     popovers.hide_all();
-    build_user_sidebar();
-    assert(user_cursor !== undefined);
-    user_cursor.reset();
+    void (async () => {
+        // We need to make sure we have all subscribers before displaying
+        // users during search, because we show all matching users and
+        // sort them by if they're subscribed.
+        $("#buddy_list_wrapper").hide();
+        loading.make_indicator($("#buddy-list-loading-subscribers"));
+        const stream_id = narrow_state.stream_id(narrow_state.filter(), true);
+        if (stream_id && !peer_data.has_full_subscriber_data(stream_id)) {
+            await peer_data.maybe_fetch_stream_subscribers(stream_id);
+        }
+        loading.destroy_indicator($("#buddy-list-loading-subscribers"));
+        $("#buddy_list_wrapper").show();
+        build_user_sidebar();
+        assert(user_cursor !== undefined);
+        user_cursor.reset();
+    })();
```

Comment:
I feel like it'd be better if you computed `stream_id` outside the `void` block, and had a bigger block of code inside the `if !peer_data.has_full_subscriber_data(stream_id)`  -- we don't need to create/destroy a loading spinner if we have the data already.

Reply:
Great point, updated